server/src/main/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapper.java

```diff
@@ -384,11 +396,44 @@ public void parseKnnVectorAndIndex(DocumentParserContext context, DenseVectorFie
                                 + "];"
                         );
                     }
-                    vector[index++] = (byte) value;
+                    consumer.accept((byte) value, index++);
```

Comment:
Why do we need to be so tricky with a `Consumer` here and have a `ByteBuffer` and `byte[]` path? As far as I understand, all our `ByteBuffer` are array-backed anyway? Can't we do without the indirection and non-static callsite here and just always insert into an array?

Reply:
Thanks for taking a look and at the suggestion @original-brownbear ! Updated to remove the `Consumer` overhead and parsing directly to a `byte` array.